runtime/src/bank/partitioned_epoch_rewards/mod.rs

```diff
@@ -150,7 +150,33 @@ pub(super) struct CalculateRewardsAndDistributeVoteRewardsResult {
 
 pub(crate) type StakeRewards = Vec<StakeReward>;
 
+pub struct KeyedRewardsAndNumPartitions {
+    pub keyed_rewards: Vec<(Pubkey, RewardInfo)>,
+    pub num_partitions: Option<u64>,
+}
+
+impl KeyedRewardsAndNumPartitions {
+    pub fn is_empty(&self) -> bool {
+        self.keyed_rewards.is_empty() && self.num_partitions.is_none()
+    }
+}
+
 impl Bank {
+    pub fn get_rewards_and_num_partitions(&self) -> KeyedRewardsAndNumPartitions {
+        let rewards = self.rewards.read().unwrap();
```

Comment:
I'd prefer an immediate clone in order to not hold onto the read lock any longer than necessary